packages/headless/src/features/commerce/facets/facet-set/facet-set-slice.test.ts

```diff
@@ -637,6 +689,261 @@ describe('facet-set slice', () => {
+      {
+        title:
+          'dispatching #toggleSelectDateFacetValue with a registered facet id',
+        facetValueState: 'selected' as FacetValueState,
+        toggleAction: toggleSelectDateFacetValue,
+      },
+      {
+        title:
+          'dispatching #toggleExcludeDateFacetValue with a registered facet id',
+        facetValueState: 'excluded' as FacetValueState,
+        toggleAction: toggleExcludeDateFacetValue,
+      },
+    ])('$title', ({facetValueState, toggleAction}) => {
+      const facetId = '1';
+      const oppositeFacetValueState = facetValueStates.find(
+        (valueState) => ![facetValueState, 'idle'].includes(valueState)
+      );
+      describe('when the facet value exists', () => {
+        it(`sets the state of an idle value to ${facetValueState}`, () => {
+          const facetValue = buildMockCommerceDateFacetValue({
+            start: '2023-01-01',
+            end: '2024-01-01',
+            endInclusive: true,
+          });
+          const facetValueRequests = convertToDateRangeRequests([facetValue]);
+
+          state[facetId] = buildMockCommerceFacetSlice({
+            request: buildMockCommerceFacetRequest({
+              type: 'dateRange',
+              values: facetValueRequests,
+            }),
+          });
+
+          const action = toggleAction({
+            facetId,
+            selection: facetValue,
+          });
+          const finalState = commerceFacetSetReducer(state, action);
+
+          const targetValue = (
+            finalState[facetId]?.request.values as DateRangeRequest[]
+          ).find(
+            (req) =>
+              req.start === facetValue.start &&
+              req.end === facetValue.end &&
+              req.endInclusive === facetValue.endInclusive
+          );
+          expect(targetValue?.state).toBe(facetValueState);
+        });
+
+        it(`sets the state of an ${oppositeFacetValueState} value to ${facetValueState}`, () => {
+          const facetValue = buildMockCommerceDateFacetValue({
+            start: '2023-01-01',
+            end: '2024-01-01',
+            endInclusive: true,
+            state: oppositeFacetValueState,
+          });
+          const facetValueRequests = convertToDateRangeRequests([facetValue]);
+
+          state[facetId] = buildMockCommerceFacetSlice({
+            request: buildMockCommerceFacetRequest({
+              type: 'dateRange',
+              values: facetValueRequests,
+            }),
+          });
+
+          const action = toggleAction({
+            facetId,
+            selection: facetValue,
+          });
+          const finalState = commerceFacetSetReducer(state, action);
+
+          const targetValue = (
+            finalState[facetId]?.request.values as DateRangeRequest[]
+          ).find(
+            (req) =>
+              req.start === facetValue.start &&
+              req.end === facetValue.end &&
+              req.endInclusive === facetValue.endInclusive
```

Comment:
This specific pattern to compare facet values for equality is used in a bunch of places in the facet code. We should isolate it to an util function IMO.
```ts
const findBy<T> = 
    (facetValue: {start: T, end: T, endInclusive: boolean}) => 
        (req: {start: T, end: T, endInclusive: boolean}) =>
          req.start === facetValue.start &&
          req.end === facetValue.end &&
          req.endInclusive === facetValue.endInclusive
```
We could then use it like this:
```ts
const targetValue = (finalState[facetId]?.request.values as DateRangeRequest[]).find(findBy(facetValue));
```

Reply:
Good idea! I added a "findExactRangeValue" method to do that